herbert_core/utilities/classes/@object_lookup/object_lookup.m

```diff
@@ -149,484 +182,374 @@
+            %   val     Index array(s) into the unique object store of the
+            %           stored object array(s):
+            %           - Numeric array if there is a single stored object array
+            %           - Cell array of index arrays, where val{i} is the vector
+            %             of indices into the store for the ith object array
+            %
+            % NOTE:
+            % - val will be stored as a column callarray of column vectors,
+            %   which is how it will be returned by the corresponding getter:
+            %       >> val = obj.indx
+            
+            % It is valid for an index array to be empty: this corresponds to an
+            % empty stored object array
+            
+            % The following permits empty index arrays to pass the test, thereby
+            % permitting the default object (i.e. constructor called with no
+            % arguments) to pass.
+            if ~iscell(val)
+                val = {val};    % turn into a cell array with one element
+            end
+            positive_int_array = @(x)(all(x>=1,'all') && all(rem(x,1)==0, 'all'));
+            if ~all (cellfun (@isnumeric, val), 'all') ||...
+                    ~all (cellfun (@(x)positive_int_array(x), val), 'all')
+                error('HERBERT:object_lookup:invalid_argument', ...
+                    'Property ''indx'' must be a cell array of integer arrays')
+            end
+            
+            % Make the set of indices a column cell array of column vectors
+            % Note x(:) makes any empty array have size (0,1), as desired for
+            % this property for any empty object array
+            obj.indx_ = cellfun (@(x)(x(:)), val(:), 'uniformOutput', false); 
+            
+            % Check interdependencies
+            if obj.do_check_combo_arg_
+                obj = obj.check_combo_arg();
+            end
+        end
+        
+        
+        function obj = set.sz (obj, val)
+            % Set the array sizes for the object arrays in the object_store
+            % Use this setter to reshape the stored object arrays
+            %
+            %   >> obj.sz = sz
+            %
+            %   sz      The size(s) of the stored object array(s) as would be 
+            %           returned by the Matlab size function
+            %           - Row vector if a single stored object array
+            %           - Cell array of row vectors where sz{i} is the Matlab
+            %             array size of the ith stored object array
+            
+            % Stored object arrays can have zero size
+            
+            if ~iscell(val)
+                val = {val};    % turn into a cell array with one element
+            end
+            
+            size_array = @(x)(isrow(x) && numel(x)>=2 && all(x>=0) && all(rem(x,1)==0));
+            if ~all (cellfun (@isnumeric, val), 'all') ||...
+                    ~all (cellfun (@(x)size_array(x), val), 'all')
```

Comment:
See `isindex`

Reply:
Earlier comment re isindex not doing precisely what I want also applies here